src/main/cpp/src/hive_hash.cu

```diff
@@ -210,11 +210,72 @@ class hive_device_row_hasher {
       return this->hash_functor.template operator()<T>(col, row_index);
     }
 
+    struct StackElement{
+      cudf::column_device_view col;   // current column
+      int child_idx;                  // index of the child column to process next
+      int factor_exp;                 // factor exponent for the current column
+      //factor_exp = parent.factor_exp + parent.child_num - 1 - parent.child_idx
+
+      __device__ StackElement() = delete;
+      __device__ StackElement(cudf::column_device_view col, int factor_exp) : col(col), child_idx(0), factor_exp(factor_exp) {}
+    };
+
+    typedef StackElement* StackElementPtr;
     template <typename T, CUDF_ENABLE_IF(cudf::is_nested<T>())>
     __device__ hive_hash_value_t operator()(cudf::column_device_view const& col,
                                             cudf::size_type row_index) const noexcept
     {
-      CUDF_UNREACHABLE("Nested type is not supported");
+      hive_hash_value_t ret = HIVE_INIT_HASH;
```

Comment:
Still better to have a detailed description on how to calculate hive hash for nested type.